app/models/order_import.rb

```diff
@@ -131,8 +131,16 @@ def handle_save_nothing_on_error # TODO refactor and rename
   end
 
   def import_row(row) # TODO refactor
-    row_importer = OrderRowImporter.new(row, self)
-    row_importer.import
+    begin
+      row_importer = OrderRowImporter.new(row, self)
+      row_importer.import
+    rescue => e
+      ActiveRecord::Base.logger.error("Failed to bulk import: #{upload_file_path}")
+      ActiveRecord::Base.logger.error(e.backtrace)
+      ExceptionNotifier.notify_exception(e, data: { message: "Failed to bulk import: #{upload_file_path}" })
+      row_importer.add_error("Failed to import row")
+      set_error_mode
```

Comment:
Hmm, I feel like this would be a very specific method that wouldn't be able to used anywhere else and this class already needs a decent amount of cognitive overhead to digest all the sub-methods. This is exception logic, so maybe the upside would be that we could move it out of the way of the normal logic.

Thoughts?